### DISCUSSION TITLE
Create NIF of the library onvif-rs

### DISCUSSION THREAD

**Discussion by luchotiz** [rusterlium/rustler#521] [Q&A]
Hi there, I was trying to create the NIF's for the onvif-rs library but I'm not sure that I'm doing the correct thing.
I get this error:
== Compilation error in file lib/onvifphx/OnvifrsNif.ex ==
** (Protocol.UndefinedError) protocol Enumerable not implemented for nil of type Atom
    (elixir 1.14.3) lib/enum.ex:1: Enumerable.impl_for!/1
    (elixir 1.14.3) lib/enum.ex:166: Enumerable.reduce/3
    (elixir 1.14.3) lib/enum.ex:4307: Enum.filter/2
    (rustler 0.27.0) lib/rustler/compiler/config.ex:121: Rustler.Compiler.Config.gather_local_crates/4
    (rustler 0.27.0) lib/rustler/compiler/config.ex:99: Rustler.Compiler.Config.external_resources/3
    (rustler 0.27.0) lib/rustler/compiler/config.ex:70: Rustler.Compiler.Config.build/1
    (rustler 0.27.0) lib/rustler/compiler.ex:9: Rustler.Compiler.compile_crate/2
    lib/onvifphx/OnvifrsNif.ex:4: (module)

Any ideas?

Thanks!

**Comment by evnu** [comment]
Can you post your `Cargo.toml`? The `Rustler.Compiler.Config` module aborts at this point:

```
    local_deps =
      current_spec["dependencies"]
      |> Enum.filter(&local_crate?/1)
      |> Enum.reject(fn dep -> MapSet.member?(visited, dep["name"]) end)
```

I suspect that `current_spec` is missing a `dependencies` section (if that is even possible).
